File: solutions/csharp/darts/1/Darts.cs
using System;

public static class Darts
{
    public static int Score(double x, double y)
    {
        double r = Math.Sqrt(Math.Pow(x, 2) + Math.Pow(y, 2));
        if (r > 10)
        {
            return 0;
        }
        else if (r > 5)
        {
            return 1;
        }
        else if (r > 1)
        {
            return 5;
        }
        else
        {
            return 10;
        }
    }
}

File: solutions/fsharp/darts/1/Darts.fs
module Darts

let score (x: double) (y: double): int =
    let distanceToCenter = sqrt (x ** 2 + y ** 2)
    match distanceToCenter with
        | r when r <= 1 -> 10
        | r when r <= 5 -> 5
        | r when r <= 10 -> 1
        | _ -> 0